src/libcore/num/mod.rs

```diff
@@ -2504,6 +2504,30 @@ impl fmt::Display for TryFromIntError {
     }
 }
 
+#[unstable(feature = "try_from", issue = "33417")]
+impl TryFrom<bool> for bool {
+    type Error = ();
```

Comment:
@cramertj Breaking change for an unstable feature is fine 😃. (Although I recommend the entire PR be held off 😛.) The blanket impl [still won't use `!` directly](https://github.com/rust-lang/rust/issues/33417#issuecomment-299124605) to avoid being blocked by `never_type`.